about_regular_expressions.exs

```diff
@@ -0,0 +1,41 @@
+defmodule AboutRegularExpressions do
+  use Koans
+
+  think "A simple regular expression" do
+    regex = ~r/This [W|w]orks!/
+
+    completion = __?
+
+    assert ("This W" <> completion) =~ regex
+    assert ("This w" <> completion) =~ regex
+  end
+
+  think "Regular expressions can use options" do
+    regex = ~r/this works!/i
+
+    assert_? "this works!" =~ regex
+    assert_? "ThIs wOrKs!" =~ regex
+  end
+
+  think "Regular expressions don't have to be created with the sigil (though ~r/.../ is more idiomatic)" do
+    regex = Regex.compile!(__?)
+
+    assert regex == ~r/this works!/
+  end
+
+  think "Sigils have flexible delimiters" do
+    assert __? == ~r(this works!)
+  end
+
+  think "Capture groups" do
+    regex = ~r/(A|The) lazy bear slept for (\d+) months/
+    text = "The lazy bear slept for 5 months"
+    assert Regex.run(regex, text) == __?
```

Comment:
We might want to give them a hint about the structure that is returned.

```elixir
assert Regex.run(regex, text) == [__?, __?]
```

Also I noticed that there are some additional options to this function (e.g. `return: :index`). Should we illustrate that usage as well?